src/SpeedDial/SpeedDial.js

```diff
@@ -0,0 +1,239 @@
+        this.setState({ nextKey: key });
+        this.setState({ prevKey: key === 'up' ? 'down' : 'up' });
+      }
+      // Select the previous action or SpeedDial
+    } else if (key === this.state.prevKey) {
+      event.preventDefault();
+      if (currentFocus.parentElement.previousElementSibling) {
+        currentFocus.parentElement.previousElementSibling.firstChild.focus();
+      } else {
+        findDOMNode(this.fab).focus();
+      }
+      // Select the next action
+    } else if (key === this.state.nextKey) {
+      event.preventDefault();
+      if (currentFocus.parentElement.nextElementSibling) {
+        currentFocus.parentElement.nextElementSibling.firstChild.focus();
+      }
+      // Close the SpeedDial
+    } else if (key === 'esc') {
+      fab.focus();
+      this.props.onClose(event, key);
+    }
+    // Forward the event
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(event, key);
+    }
+  };
+
+  render() {
+    const {
+      ButtonProps,
+      children: childrenProp,
+      classes,
+      className: classNameProp,
+      hidden,
+      icon,
+      label,
+      onClose,
+      onKeyDown,
+      open,
+      openIcon,
+      SpeedDialIconProps,
+      transition: Transition,
+      transitionDuration,
+      ...other
+    } = this.props;
+
+    const rootClassName = classNames(classes.root, classNameProp);
+    const actionsClassName = classNames(classes.actions, { [classes.actionsClosed]: !open });
+
+    // Filter the label for valid id characters.
+    const id = label.replace(/^[^a-z]+|[^\w:.-]+/gi, '');
+
+    let totalValidChildren = 0;
+    React.Children.forEach(childrenProp, child => {
+      if (React.isValidElement(child)) totalValidChildren += 1;
+    });
+
+    let validChildCount = 0;
+    const children = React.Children.map(childrenProp, child => {
+      if (!React.isValidElement(child)) return null;
+      const delay = 30 * (open ? validChildCount : totalValidChildren - validChildCount);
+      validChildCount += 1;
+      return React.cloneElement(child, {
+        delay,
+        open,
+        onKeyDown: this.handleKeyDown,
+        id: `${id}-item-${validChildCount}`,
+      });
+    });
+
+    return (
+      <Transition
+        in={!hidden}
+        className={rootClassName}
+        timeout={transitionDuration}
+        unmountOnExit
+        {...other}
+      >
+        <div>
+          <Button
+            fab
+            color="primary"
+            onKeyDown={this.handleKeyDown}
+            aria-label={label}
+            aria-haspopup="true"
+            aria-expanded={open ? 'true' : 'false'}
+            aria-controls={`${id}-actions`}
+            ref={fab => {
+              this.fab = fab;
+            }}
+            data-mui-test="SpeedDialButton"
+            {...ButtonProps}
+          >
+            <SpeedDialIcon icon={icon} openIcon={openIcon} open={open} {...SpeedDialIconProps} />
```

Comment:
What about reversing the abstraction: exposing `SpeedDialIcon` in the demos instead of the `SpeedDialIconProps` property? So people keep full control over what they add.

Reply:
Okay.